ftp/src/main/scala/akka/stream/alpakka/ftp/javadsl/FtpApi.scala

```diff
@@ -238,4 +238,18 @@ sealed trait FtpApi[FtpClient] { _: FtpSourceFactory[FtpClient] =>
 
 object Ftp extends FtpApi[FTPClient] with FtpSourceParams
 object Ftps extends FtpApi[FTPClient] with FtpsSourceParams
-object Sftp extends FtpApi[SSHClient] with SftpSourceParams
+class Sftp extends FtpApi[SSHClient] with SftpSourceParams
+
+object Sftp extends Sftp {
+
+  /**
+    * Java API: creates a [[akka.stream.alpakka.ftp.javadsl.Sftp]]
+    *
+    * @param customSshClient custom ssh client
+    * @return A [[akka.stream.alpakka.ftp.javadsl.Sftp]]
+    */
+  def withSshClient(customSshClient: SSHClient): Sftp =
```

Comment:
The `withXyz` methods are used have something to replace Scala's copy methods on case classes.
As constructor I'd recommend `create` where `apply` is used in Scala.